pandas/core/indexing.py

```diff
@@ -1844,7 +1844,8 @@ def _setitem_with_indexer_missing(self, indexer, value):
                     if len(value) != len(self.obj.columns):
                         raise ValueError("cannot set a row with mismatched columns")
 
-                value = Series(value, index=self.obj.columns, name=indexer)
+                dtype = object if isinstance(value, dict) else None
```

Comment:
i would integrate this with the if clause above

Reply:
or rather the if/else on L1840

Reply:
updated + green